File: 1342-number-of-steps-to-reduce-a-number-to-zero/1342-number-of-steps-to-reduce-a-number-to-zero.js
// JS

// A. modulo
/**
 * @param {number} num
 * @return {number}
 */
var numberOfSteps = function(num) {
    let step = 0;

    while (num) {
        num % 2 ? num -= 1 : num /= 2;
        step++;
    }

    return step;
};

// B. bitwise operations
/**
 * @param {number} num
 * @return {number}
 */
// var numberOfSteps = function(num) {
//     let step = 0;

//     while (num) {
//         num & 1 ? num -= 1 : num >>= 1;
//         step++;
//     }

//     return step;
// };

File: 1342-number-of-steps-to-reduce-a-number-to-zero/1342-number-of-steps-to-reduce-a-number-to-zero.py

# ----------------------------
# Javascript
# ----------------------------



# ----------------------------
# Python3
# ----------------------------

# A. modulo
# class Solution:
#     def numberOfSteps(self, num: int) -> int:
#         step = 0

#         while num:
#             if num % 2:
#                 num -= 1
#             else:
#                 num //= 2
#             step += 1

#         return step

# B. bitwise operations
class Solution:
    def numberOfSteps(self, num: int) -> int:
        step = 0

        while num:
            if num & 1:
                num -= 1
            else:
                num >>= 1
            step += 1

        return step
